h/features/models.py

```diff
@@ -105,3 +112,57 @@ def remove_old_flags(cls):
 
     def __repr__(self):
         return '<Feature {f.name} everyone={f.everyone}>'.format(f=self)
+
+
+class Cohort(Base, mixins.Timestamps):
+    __tablename__ = 'cohort'
+
+    id = sa.Column(sa.Integer, autoincrement=True, primary_key=True)
+    name = sa.Column(sa.UnicodeText(), nullable=False, index=True)
+
+    # Cohort membership
+    members = sa.orm.relationship(
+        'User', secondary='user_cohort', backref=sa.orm.backref(
+            'cohorts', order_by='Cohort.name'))
+
+    def __init__(self, name, creator):
+        self.name = name
+        self.creator = creator
+        self.members.append(creator)
+
+    @sa.orm.validates('name')
+    def validate_name(self, key, name):
+        if not COHORT_NAME_MIN_LENGTH <= len(name) <= COHORT_NAME_MAX_LENGTH:
+            raise ValueError('name must be between {min} and {max} characters '
+                             'long'.format(min=COHORT_NAME_MIN_LENGTH,
+                                           max=COHORT_NAME_MAX_LENGTH))
+        return name
+
+    def __repr__(self):
+        return '<Cohort: %s>' % self.slug
+
+    @property
+    def slug(self):
+        """A version of this group's name suitable for use in a URL."""
+        return slugify.slugify(self.name)
+
+    @classmethod
+    def get_by_id(cls, id_):
```

Comment:
You can just use SQLAlchemy's [get](http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.get) method which returns `None` when it can't be found.

And we also recently tried to pass in the database session to class methods on models to correctly scope these queries in the current transaction like [here](https://github.com/hypothesis/h/blob/master/h/api/models/document.py#L43-L55), which in this case would look like: `session.query(cls).get(id_)`